### PR TITLE
adjustments to webpack to improve build speed

### PR DESCRIPTION
**Pull request by BSd3v**
adjustments to webpack to improve build speed

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -57,12 +58,15 @@ module.exports = (env, argv) => {
         module: {
             rules: [
                 {
+                    exclude: /node_modules/,
```
Comment (ndrezn) [inline webpack.config.js:61]: @alexcjohnson, curious what your thought on this exclusion is?

**Comment by AnnMarieW** [comment]
One more minor (and unrelated) item I found when I was updating the docs recently:

Console msg:
> ag-grid-community.auto.esm.js:484 AG Grid: setDatasource is deprecated. Please use 'api.setGridOption('datasource', newValue)' or 'api.updateGridOptions({ datasource: newValue })' instead.


https://github.com/plotly/dash-ag-grid/blob/97b4830a34a7e493e25ced31ca841a174b0df3ac/src/lib/fragments/AgGrid.react.js#L913

Do you think you could include this small update in this PR?

**Comment by BSd3v** [comment]
Oh. Interesting. 

Is it to be declared as a prop? The name didn't change...

**Comment by AnnMarieW** [comment]
It would be nice to still handle this in the component rather than the dash app.  Just needs to be set in the `gridOptions` ?

**Comment by BSd3v** [comment]
Yeah. It would be in the component.

**Review by ndrezn — APPROVED** [review]
🚀 Looks great. 

I am seeing:
```
tests/test_cell_data_type_override.py::test_cd001_cell_data_types_override FAILED                      [  5%]

E           selenium.common.exceptions.TimeoutException: Message: text -> 17/01/2024 not found within 10s, found: 04/02/11720
```
and:
```
tests/test_popupparent.py::test_pp001_popupParent FAILED                                               [ 60%]

E       selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":"body > .ag-popup .mantine-Select-input"}
```
failing locally but I believe this is unrelated to this PR.

This is a great change! I'm seeing on my M1 Pro:
```
npm run build  10.78s user 1.12s system 167% cpu 7.114 total
```

And a brutal:
```
npm run build  233.24s user 13.03s system 130% cpu 3:09.10 total
```

on `main`. Good stuff thank you @BSd3v!

**Pull request merged**